### ISSUE TITLE
phpstan complains about Http\Client\Exception

### ISSUE THREAD

**Issue by ramsey** [php-http/httplug#142]
| Q            | A
| ------------ | ---
| Bug?         | maybe
| New Feature? | no
| Version      | 1.1.0

#### Actual Behavior

When building an HTTP client that implements `Http\Client\HttpClient`, I am implementing the `sendRequest()` method, as required by the interface. I have added the `@throws` docblock annotation with `Http\Client\Exception`, since that is what the interface defines. When I run `phpstan`, I receive the following error because `phpstan` detects that the `Exception` interface does not inherit from `Throwable`.

```
$ composer run phpstan
> phpstan analyse MyClient.php --level max --no-progress
 ------ ----------------------------------------------------------------------
  Line   MyClient.php
 ------ ----------------------------------------------------------------------
  32     PHPDoc tag @throws with type Http\Client\Exception is not subtype of
         Throwable
 ------ ----------------------------------------------------------------------

 [ERROR] Found 1 error
```

#### Expected Behavior

I expect static analysis tools to pass without error when defining `@throws \Http\Client\Exception` on a method docblock.

#### Steps to Reproduce

1. Use the `composer.json` and `MyClient.php` files provided below.
2. Run `composer install`
3. Run `composer run phpstan`

##### composer.json

``` json
{
    "require": {
        "php-http/httplug": "^1.1",
        "phpstan/phpstan": "^0.10.3"
    },
    "autoload": {
        "psr-4": {
            "Ramsey\\Example\\": "."
        }
    },
    "scripts": {
        "phpstan": "phpstan analyse MyClient.php --level max --no-progress"
    }
}
```

##### MyClient.php

``` php
<?php

declare(strict_types=1);

namespace Ramsey\Example;

use Http\Client\Exception;
use Http\Client\HttpClient;
use Psr\Http\Message\RequestInterface;
use Psr\Http\Message\ResponseInterface;

class MyClient implements HttpClient
{
    /**
     * @var HttpClient
     */
    private $client;

    /**
     * @param HttpClient $client
     */
    public function __construct(HttpClient $client)
    {
        $this->client = $client;
    }

    /**
     * @param RequestInterface $request
     * @return ResponseInterface
     * @throws Exception
     */
    public function sendRequest(RequestInterface $request): ResponseInterface
    {
        return $this->client->sendRequest($request);
    }
}
```

#### Possible Solutions

The `Http\Client\Exception` interface should extend `Throwable`, which would cause `phpstan` and other static analysis tools to run without errors.

I understand this may not be possible, since this library supports versions of PHP earlier than 7.0. This is also why I put "maybe" as an answer to whether this is a bug.

**Comment by ramsey** [comment]
I've created two separate pull requests that implement a solution for this issue, using two different approaches: #143 and #144.

**Comment by Nyholm** [comment]
Thank you. See #145

**Comment by ramsey** [comment]
Feel free to close this and the PRs, since it’s being addressed. I can add an ignore pattern to phpstan in the meantime.

**Comment by dbu** [comment]
thanks for the report. agree with @Nyholm that version 2 of httplug should drop the obsolete PHP versions to be able to do things correctly.

meanwhile, can phpstan be told that some codebase is PHP 5 and that it should not ask for things that are not possible in PHP 5?

**Comment by ramsey** [comment]
phpstan has regular expression matching to ignore certain errors, so that’s what I’m doing for now.

**Comment by dbu** [comment]
fixed in #145, php-http 2.0 is about to be released
